go-controller/pkg/ovn/egressip.go

```diff
@@ -1268,43 +1652,137 @@ func (e *EgressIPController) syncStaleSNATRules(egressIPCache map[string]egressI
 // atomic items with the same general information repeated across most (egressIP
 // name, logical IP defined for that name), hence use a cache to avoid round
 // trips to the API server per item.
-func (e *EgressIPController) generateCacheForEgressIP() (map[string]egressIPCacheEntry, error) {
-	egressIPCache := make(map[string]egressIPCacheEntry)
-	egressIPs, err := e.watchFactory.GetEgressIPs()
+func (e *EgressIPController) generateCacheForEgressIP() (egressIPCache, error) {
```

Comment:
If the UDN is deleted we dont have to worry as the config will be removed.
If the UDN is present, I think its ok because the secondary controllers dont depend on the EIP controller to be started. 

> or does this work because egress IP controller is not started until the retry framework starts in DNC? If so, do we end up missing any events or scenarios when secondary network controllers have already come up and started.

I cannot think of any.

Reply:
its also possible i fully extract the EIP controller from DNC and avoid all this. The code is setup for that now, just a little more effort tim.

Reply:
IMHO it would be preferable to extract the EIP controller for the default network ... 

But I don't understand how much more effort it would take, and if that can be squeezed into your deadline ...

Reply:
"If the UDN is present, I think its ok because the secondary controllers dont depend on the EIP controller to be started."

So I see the secondary l3 controller calls: ensureL3ClusterRouterPoliciesForNetwork, which relies on nodeZoneState cache, which is updated during syncEgressIPs:

```
	// update localZones cache of eIPCZoneController
	// WatchNodes() is called before WatchEgressIPNamespaces() so the oc.localZones cache
	// will be updated whereas WatchEgressNodes() is called after WatchEgressIPNamespaces()
	// and so we must update the cache to ensure we are not stale.
	if err := e.syncLocalNodeZonesCache(); err != nil {
		return fmt.Errorf("syncLocalNodeZonesCache unable to update the local zones node cache: %v", err)
	}
```

I think we need to fix this. If its easy for you to rip out the EIP controller from DNC and start it separately that is good. Otherwise we can explore just abstracting out the call to the sync function to run before the controllers are started, and remove it from syncing in the DNC retry framework. That should be very easy to do.

Reply:
I apologise for that comment to you. I should have thought about that interaction between the two components.
I explored both paths and I would like to extract the EIP controller fully from the DNC in a follow up PR.
I added a commit on top to sync the OVN zone cache.